yarp/yarp_compiler.c

```diff
@@ -1335,6 +1335,45 @@ yp_compile_node(rb_iseq_t *iseq, const yp_node_t *node, LINK_ANCHOR *const ret,
 
           return;
       }
+      case YP_NODE_INTERPOLATED_X_STRING_NODE: {
+          yp_interpolated_x_string_node_t *interp_x_string_node = (yp_interpolated_x_string_node_t *) node;
+          size_t parts_size = interp_x_string_node->parts.size;
+
+          ADD_INSN(ret, &dummy_line_node, putself);
+
+          if (parts_size > 0) {
+              for (size_t index = 0; index < parts_size; index++) {
+                  yp_node_t *part = interp_x_string_node->parts.nodes[index];
+
+                  switch (part->type) {
```

Comment:
Can you use `YP_NODE_TYPE` here? That will make it so that the switch will be exhaustive.